From: NeilBrown <neilb@suse.com>
Subject: KABI fix
Patch-mainline: Never, kabi
References: bsc#1014410

If this struct is used by an external module, it might
embed it in a larger struct, so changing size it not acceptable.
Fortunately:
1/ we only need about 30 bits.  This number is at most as few
   minutes, measured in jiffies.  30bits gives 1 fortnight
   at 1024Hz.
2/ There are several 4-bytes holes on 64bit archs, but on
   32, there is only a 30bit bit-field.

So make the field an 'int', and move it to the hole.

Acked-by: NeilBrown <neilb@suse.com>
---
 include/linux/sunrpc/xprt.h |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

--- a/include/linux/sunrpc/xprt.h
+++ b/include/linux/sunrpc/xprt.h
@@ -173,6 +173,9 @@ struct rpc_xprt {
 	unsigned long		state;		/* transport state */
 	unsigned char		shutdown   : 1,	/* being shut down */
 				resvport   : 1; /* use a reserved port */
+#ifndef __GENKSYMS__
+	unsigned int		max_reconnect_timeout:30;
+#endif
 	unsigned int		swapper;	/* we're swapping over this
 						   transport */
 	unsigned int		bind_index;	/* bind function index */
@@ -192,8 +195,7 @@ struct rpc_xprt {
 	struct work_struct	task_cleanup;
 	struct timer_list	timer;
 	unsigned long		last_used,
-				idle_timeout,
-				max_reconnect_timeout;
+				idle_timeout;
 
 	/*
 	 * Send stuff
